judgments/views/advanced_search.py

```diff
@@ -1,179 +1,188 @@
-from django.utils.translation import gettext
-from ds_caselaw_utils import courts as all_courts
+from django.utils.translation import gettext as _
+from django.views.decorators.csrf import csrf_exempt
 
+from judgments.forms import AdvancedSearchForm
 from judgments.models.court_dates import CourtDates
-from judgments.models.search_form_errors import SearchFormErrors
 from judgments.utils import (
     MAX_RESULTS_PER_PAGE,
     api_client,
     as_integer,
     has_filters,
     paginator,
-    parse_date_parameter,
-    preprocess_query,
     process_court_facets,
     process_year_facets,
     show_no_exact_ncn_warning,
 )
 
 
-def advanced_search(request):
-    params = request.GET
-    errors = SearchFormErrors()
-    try:
-        from_date = parse_date_parameter(
-            params,
-            "from",
-            start_year=CourtDates.min_year(),
-            end_year=CourtDates.max_year(),
-        )
-    except ValueError as error:
-        from_date = None
-        errors.add_error(
-            gettext("search.errors.from_date_headline"), "from_date", str(error)
-        )
-    try:
-        to_date = parse_date_parameter(
-            params,
-            "to",
-            default_to_last=True,
-            start_year=CourtDates.min_year(),
-            end_year=CourtDates.max_year(),
-        )
-    except ValueError as error:
-        to_date = None
-        errors.add_error(
-            gettext("search.errors.to_date_headline"), "to_date", str(error)
-        )
-
-    if to_date is not None and from_date is not None and to_date < from_date:
-        errors.add_error(
-            gettext("search.errors.to_before_from_headline"),
-            "to_date",
-            gettext("search.errors.to_before_from_detail"),
-        )
-
-    query_params = {
-        "query": params.get("query", ""),
-        "court": params.getlist("court"),
-        "judge": params.get("judge"),
-        "party": params.get("party"),
-        "order": params.get("order", ""),
-        "from": from_date,
-        "from_day": params.get("from_day"),
-        "from_month": params.get("from_month"),
-        "from_year": params.get("from_year"),
-        "to": to_date,
-        "to_day": params.get("to_day"),
-        "to_month": params.get("to_month"),
-        "to_year": params.get("to_year"),
-        "per_page": params.get("per_page"),
-    }
-
-    query_text = query_params["query"]
-    page = str(as_integer(params.get("page"), minimum=1))
-    per_page = str(
-        as_integer(
-            params.get("per_page"),
-            minimum=1,
-            maximum=MAX_RESULTS_PER_PAGE,
-            default=RESULTS_PER_PAGE,
-        )
-    )
-
-    order = query_params["order"]
-    # If there is no query, order by -date, else order by relevance
-    if not order and not query_text:
-        order = "-date"
-    elif not order:
-        order = "relevance"
+def _do_dates_require_warnings(from_date):
+    from_warning = False
```

Comment:
docstring, please!

Reply:
"""Warn the user if they have specified a from date in their search and it's before the first records we have for any court."""